### PR TITLE
Test swift-models.

### PR DESCRIPTION
**Pull request by pschuh**
Basic testing of building swift-models in addition to swift-apis

### CODE REVIEW

File: Sources/TensorFlow/Loss.swift [hunk 1]
```diff
@@ -194,6 +194,14 @@ public func kullbackLeiblerDivergence<Scalar: TensorFlowFloatingPoint>(
     reduction(expected * log(expected / predicted))
 }
 
+/// Workaround for cross-module default parameter @differentiable functions.
+@differentiable
+public func _justTakesTheMean<Scalar: TensorFlowFloatingPoint>(
```
Comment (dan-zheng) [inline Sources/TensorFlow/Loss.swift]: Could you please rename this to just `mean`? If that's ambiguous, then `_mean`?
```suggestion
public func mean<Scalar: TensorFlowFloatingPoint>(
```

File: Sources/TensorFlow/Loss.swift [hunk 2]
```diff
@@ -52,7 +52,7 @@ public func meanAbsoluteError<Scalar: TensorFlowFloatingPoint>(
     predicted: Tensor<Scalar>,
     expected: Tensor<Scalar>
 ) -> Tensor<Scalar> {
-    l1Loss(predicted: predicted, expected: expected, reduction: { $0.mean() })
```
Comment (eaplatanios) [inline Sources/TensorFlow/Loss.swift]: I believe there is also a couple of loss functions that use `sum()`. Shouldn't we modify those too?

Comment (dan-zheng) [reply Sources/TensorFlow/Loss.swift]: The PR tests integrating TensorFlow directly into a toolchain:
* Delete `TensorFlow` module and artifacts from an existing toolchain.
* Use toolchain to build `swift-apis`.
* Copy built `TensorFlow` artifacts back into the toolchain.

`_mean` is a workaround for linker errors encountered during CI for `swift-models`:
```
<compiler-generated>:0: error: undefined reference to 'AD__$s10TensorFlow19softmaxCrossEntropy6logits6labels9reductionAA0A0VyxGAH_AGys5Int32VGA2HXFtAA0aB13FloatingPointRzlFfA1_A2HcfU___jvp_src_0_wrt_0'
<compiler-generated>:0: error: undefined reference to 'AD__$s10TensorFlow19softmaxCrossEntropy6logits6labels9reductionAA0A0VyxGAH_AGys5Int32VGA2HXFtAA0aB13FloatingPointRzlFfA1_A2HcfU___vjp_src_0_wrt_0'
<compiler-generated>:0: error: undefined reference to 'AD__$s10TensorFlow19softmaxCrossEntropy6logits6labels9reductionAA0A0VyxGAH_AGys5Int32VGA2HXFtAA0aB13FloatingPointRzlFfA1_A2HcfU___pullback_src_0_wrt_0'
```

Adding `_mean` and renaming usages was sufficient to fix these errors, so `_sum` is not necessary at the moment. We're debugging these linker errors so we can remove `_mean` ASAP.

Comment (eaplatanios) [reply Sources/TensorFlow/Loss.swift]: Great, thanks Dan!

Comment (rxwei) [reply Sources/TensorFlow/Loss.swift]: Is there a bug filed for the linker error?

Comment (dan-zheng) [reply Sources/TensorFlow/Loss.swift]: Filed [TF-690](https://bugs.swift.org/browse/TF-690].
